pinot-core/src/main/java/com/linkedin/pinot/core/realtime/converter/stats/RealtimeColumnStatistics.java

```diff
@@ -0,0 +1,156 @@
+/**
+ * Copyright (C) 2014-2016 LinkedIn Corp. (pinot-core@linkedin.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.core.realtime.converter.stats;
+
+import com.linkedin.pinot.common.data.FieldSpec;
+import com.linkedin.pinot.core.common.Block;
+import com.linkedin.pinot.core.common.BlockMultiValIterator;
+import com.linkedin.pinot.core.data.partition.PartitionFunction;
+import com.linkedin.pinot.core.io.reader.SingleColumnSingleValueReader;
+import com.linkedin.pinot.core.operator.blocks.RealtimeSingleValueBlock;
+import com.linkedin.pinot.core.realtime.impl.datasource.RealtimeColumnDataSource;
+import com.linkedin.pinot.core.realtime.impl.dictionary.MutableDictionaryReader;
+import com.linkedin.pinot.core.segment.creator.ColumnStatistics;
+import java.util.List;
+import org.apache.commons.lang.math.IntRange;
+
+
+/**
+ * Column statistics for a column coming from an in-memory realtime segment.
+ */
+public class RealtimeColumnStatistics implements ColumnStatistics {
+  private final RealtimeColumnDataSource _dataSource;
+  private final int[] _sortedDocIdIterationOrder;
+  private final MutableDictionaryReader _dictionaryReader;
+  private final Block _block;
+
+  public RealtimeColumnStatistics(RealtimeColumnDataSource dataSource, int[] sortedDocIdIterationOrder) {
+    _dataSource = dataSource;
+    _sortedDocIdIterationOrder = sortedDocIdIterationOrder;
+    _dictionaryReader = dataSource.getDictionary();
+    _block = dataSource.getNextBlock();
+  }
+
+  @Override
+  public Object getMinValue() {
+    return _dictionaryReader.getMinVal();
+  }
+
+  @Override
+  public Object getMaxValue() {
+    return _dictionaryReader.getMaxVal();
+  }
+
+  @Override
+  public Object getUniqueValuesSet() {
+    return _dictionaryReader.getSortedValues();
+  }
+
+  @Override
+  public int getCardinality() {
+    return _dictionaryReader.length();
+  }
+
+  @Override
+  public int getLengthOfLargestElement() {
+    // Length of longest string
+    int maximumStringLength = 0;
```

Comment:
It's equivalent and less verbose to do it this way. If I try to set it to the length of the first element, I need to add a check for the dictionary being empty.